src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1387,14 +1387,19 @@ public static string FullName(this IEdmSchemaElement element)
             {
                 return string.Empty;
             }
-            else if (element.Namespace == null)
+
+            if (element.Namespace == null)
             {
                 return element.Name;
             }
-            else
+
+            IEdmFullNamedElement fullNamedElement = element as IEdmFullNamedElement;
+            if (fullNamedElement != null)

```

Comment:
nit: more compact to use ternary operator ?:

Reply:
It looks a bit ugly in this case

return fullNamedElement?.FullName ??  (element.Namespace + "." + element.Name);

I prefer what is existing

Reply:
I think the current code Yogesh has would be easier to read.

Reply:
I agree.


---
In reply to: [209406678](https://github.com/OData/odata.net/pull/1227#discussion_r209406678) [](ancestors = 209406678)